chain/cmd/provider/main.go

```diff
@@ -22,34 +22,78 @@ import (
 	"github.com/bandprotocol/d3n/chain/x/zoracle"
 )
 
+type Command struct {
+	Cmd       string   `json:"cmd"`
+	Arguments []string `json:"args"`
+}
+
 const limitTimeOut = 10 * time.Second
 
 // TODO: Replace by `BandStatefulClient` after implementation finished.
 var bandProvider d3nlib.BandProvider
 var allowedCommands = map[string]bool{"curl": true, "date": true}
 
-func main() {
-	// Get environment variable
-	privS, ok := os.LookupEnv("PRIVATE_KEY")
-	if !ok {
-		log.Fatal("Missing private key")
+func getEnv(key, def string) string {
+	tmp := os.Getenv(key)
+	if tmp == "" {
+		return def
 	}
-	nodeURI, ok := os.LookupEnv("NODE_URI")
-	if !ok {
-		log.Fatal("Missing node uri")
+	return tmp
+}
+
+var (
+	nodeURI  = getEnv("NODE_URI", "http://localhost:26657")
+	queryURI = getEnv("QUERY_URI", "http://localhost:1317")
+	privS    = getEnv("PRIVATE_KEY", "eedda7a96ad35758f2ffc404d6ccd7be913f149a530c70e95e2e3ee7a952a877")
+)
+
+func getLatestRequest() (uint64, error) {
+	resp, err := grequests.Get(fmt.Sprintf("%s/zoracle/request_number", queryURI), nil)
+	if err != nil {
+		return 0, err
+	}
+
+	var responseStruct struct {
+		Result string `json:"result"`
 	}
-	viper.Set("nodeURI", nodeURI)
+	if err := resp.JSON(&responseStruct); err != nil {
+		return 0, err
+	}
+
+	return strconv.ParseUint(responseStruct.Result, 10, 64)
+}
 
+func main() {
 	privB, _ := hex.DecodeString(privS)
 	var priv secp256k1.PrivKeySecp256k1
 	copy(priv[:], privB)
 
 	var err error
-	bandProvider, err = d3nlib.NewBandProvider(priv)
+	bandProvider, err = d3nlib.NewBandProvider(nodeURI, priv)
 	if err != nil {
 		panic(err)
 	}
-	s := sub.NewSubscriber(viper.GetString("nodeURI"), "/websocket")
+
+	currentRequest, err := getLatestRequest()
+	if err != nil {
+		panic(err)
+	}
+
+	// Setup poll loop
+	for {
+		newRequest, err := getLatestRequest()
+		if err != nil {
+			log.Println("Cannot get request number error: ", err.Error())
+		}
+
+		for currentRequest < newRequest {
+			currentRequest++
+			go newHandleRequest(currentRequest)
+		}
+		time.Sleep(2 * time.Second)
+	}
+
+	s := sub.NewSubscriber(nodeURI, "/websocket")
```

Comment:
Let's remove all this unreachable code and the whole sub folder?

Reply:
We haven't finished new request handler, after implement those ticket. It will be removed later